controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java

```diff
@@ -633,11 +633,20 @@ private Scope getScope(final String scopeName) {
         return getScaleMetadata(scopeName, streamName, null, executor).thenApply(scaleMetadataList -> {
             long size = scaleMetadataList.size();
             long totalNumSplits = 0;
-
-            for (int i = 0, j = 1; j < size; j++) {
-                long countSplitsEventI = scaleMetadataList.get(i).getSegments().size();
-                long countSplitsEventJ = scaleMetadataList.get(j).getSegments().size();
-                totalNumSplits = (countSplitsEventI < countSplitsEventJ) ? (countSplitsEventJ - countSplitsEventI) : 0;
+            long countSplitsEventI, countSplitsEventI1;
+
+            if (isDescendingOrder(scaleMetadataList)) {
+                for (int i = 0; i < size; i++) {
+                    countSplitsEventI = scaleMetadataList.get(i).getSegments().size();
```

Comment:
how is number of events == splits?

Note. in a scale operation you can have both splits and merges:
take following example where segments 0,1,2 result in creation of {3, 4, 5, 6} 
where:
{0, 1} merge into {3}
and {2} splits into {4, 5, 6}
so you have a split of 3 and a merge of 2 in the above example. 
thats what you need to count.. i am not sure what is being counted here. its only counting number of active segments that are there in scaleMetadataList.get(i)